### ISSUE TITLE
[Clang Tidy][C++20 Modules] Clang Tidy too slow and buggy on modules

### ISSUE THREAD

**Issue by aaronmondal** [llvm/llvm-project#63939]
Importing the `std` module seems to evaluate clang-tidy on the entire libcxx on every invocation. That's probably a bug since every file takes 10 years to evaluate, but it's also useful to test module compatibility :smile: 

The following snippet, built with C++23, crashes all warnings below, i.e. to run clang-tidy on code with `import std;` you'll currently have to disable (at least) all of these.

An outlier here is `misc-include-cleaner` which doesn't crash, but also doesn't detect that `std::` symbols come from `import std;`.

```cpp
import std;

auto main() -> int {
  constexpr std::string_view kExplanation =
      "Clang Tidy crashes a lot with modules :D";

  std::cout << std::format("{}, {}!\n{}", "Hello", "world", kExplanation)
            << '\n';
  return 0;
}
```
```
-bugprone-dangling-handle
-bugprone-fold-init-type
-bugprone-implicit-widening-of-multiplication-result
-bugprone-move-forwarding-reference
-bugprone-narrowing-conversions
-bugprone-shared-ptr-array-mismatch
-bugprone-string-integer-assignment
-bugprone-stringview-nullptr
-bugprone-swapped-arguments
-bugprone-unique-ptr-array-mismatch
-bugprone-use-after-move
-cppcoreguidelines-missing-std-forward
-cppcoreguidelines-narrowing-conversions
-cppcoreguidelines-pro-bounds-array-to-pointer-decay
-cppcoreguidelines-rvalue-reference-param-not-moved
-google-build-explicit-make-pair
-misc-const-correctness
-misc-include-cleaner (doesn't crash, but fails)
-misc-uniqueptr-reset-release
-modernize-make-shared
-modernize-make-unique
-modernize-use-nullptr
-modernize-use-transparent-functors
-performance-avoid-endl
-performance-inefficient-algorithm
-performance-inefficient-string-concatenation
-performance-move-const-arg
-performance-unnecessary-value-param
-portability-std-allocator-const
-readability-redundant-declaration
-readability-redundant-smartptr-get
-readability-redundant-string-cstr
-readability-redundant-string-init
-readability-uppercase-literal-suffix
```

**Comment by ChuanqiXu9** [comment]
Yeah, it reveals that we lack a design (and the corresponding implementation) for modules in clang-tidy. (Also the may be other tools too.)
